.github/workflows/test.yml

```diff
@@ -0,0 +1,61 @@
+name: 🐘 Test
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron:  '0 14 3 * *' # Monthly at 2pm on the third
+jobs:
+  build:
+    strategy:
+      matrix:
+        pg:
+          - { version: 14,  upgrade_to: "",  update_from: 0.99.0 }
+          - { version: 13,  upgrade_to: 14,  update_from: 0.99.0 }
+          - { version: 12,  upgrade_to: 13,  update_from: 0.99.0 }
+          - { version: 11,  upgrade_to: 12,  update_from: 0.99.0 } # Versions prior to 0.99.0 don't support Postgres 11
+          - { version: 10,  upgrade_to: 11,  update_from: 0.95.0 }
+          - { version: 9.6, upgrade_to: 10,  update_from: 0.95.0 }
+          - { version: 9.5, upgrade_to: 9.6, update_from: 0.95.0 }
+          - { version: 9.4, upgrade_to: 9.5, update_from: 0.95.0 }
+          - { version: 9.3, upgrade_to: 9.4, update_from: 0.95.0 }
+          - { version: 9.2, upgrade_to: 9.3, update_from: "" }     # updatecheck is not supported prior to 9.3
+          - { version: 9.1, upgrade_to: 9.2, update_from: "" }
+          # Also test pg_upgrade across many versions
+          # - { version: 9.1,  upgrade_to: 14, update_from: 0.99.0 }
+          # - { version: 9.4,  upgrade_to: 14, update_from: 0.99.0 }
+        parallel:
+          - { conn: "",  name: 'parallel' }
+          - { conn: "1", name: 'serial'   }
+    name: 🐘 PostgreSQL ${{ matrix.pg.version }} (${{ matrix.parallel.name }})
+    runs-on: ubuntu-latest
+    container: pgxn/pgxn-tools
+    env:
+      PGUSER: postgres
+      PARALLEL_CONN: "${{ matrix.parallel.conn }}"
+      UPDATE_FROM:  "${{ matrix.pg.update_from }}"
+    steps:
+      - run: pg-start ${{ matrix.pg.version }}
+      - uses: actions/checkout@v2
+
+      # Basic regression test.
+      - run: pg-build-test
+
+      # Test update.
+      - run: 'if [ -d "$UPDATE_FROM" ]; then make uninstall clean updatecheck; fi'
+
+      # Test upgrade.
+      # - run: ./test/test_MVU.sh -s 55667 55778 "{{ matrix.pg.version }}" "{{ matrix.pg.upgrade_to }}" "/usr/lib/postgresql/{{ matrix.pg.version }}/bin/" "/usr/lib/postgresql/{{ matrix.pg.upgrade_to }}/bin/"
+
+      # Test all, install, test, test-serial, and test-parallel, both from clean
+      # repo and repeated with existing build.
+      - run: make uninstall clean all
+      - run: make all
+      - run: make uninstall clean install
+      - run: make install
+      - run: "psql -Ec 'CREATE EXTENSION pgtap'"
+      - run: make uninstall clean test
+      - run: make test
+      - run: make uninstall clean test-serial
+      - run: make test-serial
+      - run: make uninstall clean test-parallel
```

Comment:
If we want to run both serial and parallel tests together then you should pull the parallel setting out of the matrix. Otherwise, these should be conditional based on what $PARALLEL_CONN is set to. When I wrote this stuff originally I specifically broke them out because it made it easier to debug where failures were coming from (ie, are they related to parallel or not).

Reply:
I could have sworn I replicated what you had. Will take another look.

Reply:
Is [this](https://github.com/theory/pgtap/blob/5394f34/tools/pg-travis-test.sh#L112-L116) doing a test, then a test-serial, then a test-parallel, twice each? And in the Travis Matrix, you're running those tests first [with parallelism enabled](https://github.com/theory/pgtap/blob/5394f34/.travis.yml#L8-L18) and then [with parallelism disabled](https://github.com/theory/pgtap/blob/5394f34/.travis.yml#L20-L30)?

Reply:
Thinking back... I think what I was actually doing here is making sure that test-parallel with PARALLEL_CONN=1 actually worked... which seems like overkill. In any case, I don't think what you've got here (setting `conn: 1`) will actually translate into PARALLEL_CONN=1.

BTW, note that [`test-parallel` with `PARALLEL_CONN=1`[(https://github.com/theory/pgtap/blob/5394f343d4da512c20cb80e14ad6ed2983d22479/Makefile#L320-L325) is [not the same as `test-serial`](https://github.com/theory/pgtap/blob/5394f343d4da512c20cb80e14ad6ed2983d22479/Makefile#L315-L318).

Reply:
Line 34 sets `PARALLEL_CONN`: `"${{ matrix.parallel.conn }}"` inserts the value for each iteration of the `parallel` matrix on lines 27-28,. However it occurred to me I could probably reduce the number of build by just having separate `run:` items with and without `PARALLEL_CONN=1`.

>  is not the same as test-serial

Not, but I assumed your original intent was to ensure that `test-parallel` worked with or without `PARALLEL_CONN=1`.

Reply:
What do you think of this: I remove the `parallel` matrix and then run this:

```yaml
      - run: make uninstall clean all
      - run: make all
      - run: make uninstall clean install
      - run: make install
      - run: "psql -Ec 'CREATE EXTENSION pgtap'"
      - run: make uninstall clean test
      - run: make test
      - run: PARALLEL_CONN=1 make uninstall clean test
      - run: PARALLEL_CONN=1 make test
      - run: make uninstall clean test-serial
      - run: make test-serial
      - run: PARALLEL_CONN=1 make uninstall clean test-serial
      - run: PARALLEL_CONN=1 make test-serial
      - run: make uninstall clean test-parallel
      - run: make test-parallel
      - run: PARALLEL_CONN=1 make uninstall clean test-parallel
      - run: PARALLEL_CONN=1 make test-parallel
```

Reply:
Thoughts on this, @nasbyj?